bin/pre_download_compilers

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# This ./cache/hardhat-nodejs is coming from the env-paths module
+# that hardhat is using.
+COMPILER_DIR=/root/.cache/hardhat-nodejs/compilers-v2
+mkdir -p $COMPILER_DIR/linux-amd64
+mkdir -p $COMPILER_DIR/vyper/linux
+mkdir -p $COMPILER_DIR/zksolc
+mkdir -p $COMPILER_DIR/zkvyper
+
+# Fetch latest compiler version
+wget -nv -O $COMPILER_DIR/zksolc/compilerVersionInfo.json  "https://raw.githubusercontent.com/matter-labs/zksolc-bin/main/version.json"
+
+
+# These are the versions that we currently have in hardhat.config.ts in zksync-era and era-contracts.
+# For now, if there is a new version of compiler, we'd have to modify this file.
+# In the future, we should make it more automatic.
+(for ver in v1.3.18 v1.3.21 v1.4.0 v1.4.1; do wget -nv -O $COMPILER_DIR/zksolc/zksolc-$ver  https://raw.githubusercontent.com/matter-labs/zksolc-bin/main/linux-amd64/zksolc-linux-amd64-musl-$ver; done)
+
+# Special pre-release 1.5.0 compiler.
+# It can be removed once system-contracts/hardhatconfig.ts stops using it.
+wget -nv -O $COMPILER_DIR/zksolc/zksolc-remote-4cad2deaa6801d7a419f1ed6503c999948b0d6d8.0 https://github.com/matter-labs/era-compiler-solidity/releases/download/prerelease-a167aa3-code4rena/zksolc-linux-amd64-musl-v1.5.0
+
+
+wget -nv -O $COMPILER_DIR/zkvyper/compilerVersionInfo.json  "https://raw.githubusercontent.com/matter-labs/zkvyper-bin/main/version.json"
+
+(for ver in v1.3.13; do wget -nv -O $COMPILER_DIR/zkvyper/zkvyper-$ver  https://raw.githubusercontent.com/matter-labs/zkvyper-bin/main/linux-amd64/zkvyper-linux-amd64-musl-$ver; done)
+
+
+# This matches VYPER_RELEASES_MIRROR_URL from hardhat-vyper
+wget -nv -O $COMPILER_DIR/vyper/linux/list.json https://vyper-releases-mirror.hardhat.org/list.json
+
+# Currently we only use 0.3.10 release of vyper compiler (path taken from the list.json above)
+wget -nv -O $COMPILER_DIR/vyper/linux/0.3.10 https://github.com/vyperlang/vyper/releases/download/v0.3.10/vyper.0.3.10%2Bcommit.91361694.linux
```

Comment:
why can't we use https://raw.githubusercontent.com use github cdn host for vyperlang?

Reply:
I wasn't able to make it work for releases, they are not a part of the repo, github redirects the path above to:
`https://objects.githubusercontent.com/github-production-release-asset-2e65be/73461676/c1dd89db-ee80-4888-8d99-f68488ab4331?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240529%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240529T220521Z&X-Amz-Expires=300&X-Amz-Signature=a5b1dfb06430ab5707231016f801d12baeb1aae96f3901b1ee84cec963f9871d&X-Amz-SignedHeaders=host&actor_id=18558509&key_id=0&repo_id=73461676&response-content-disposition=attachment%3B%20filename%3Dvyper.0.3.10%2Bcommit.91361694.linux&response-content-type=application%2Foctet-stream`